front/src/modules/simulationResult/components/SpaceTimeChart/index.tsx

```diff
@@ -120,8 +169,8 @@ export default function SpaceTimeChart(props: SpaceTimeChartProps) {
       if (trainSimulations) {
         const trains = trainSimulations.map((train) =>
           train.id === selectedTrain.id ? timeShiftTrain(train as Train, offset) : train
-        );
-        setTrainSimulations(trains as Train[]);
+        ) as Train[];
```

Comment:
After talking with @clarani, we decided we can directly cast in this PR. We want to make things simpler by using just one type instead of two in another PR

Reply:
We can leave it for later, I created this ticket to merge the 2 types: https://github.com/osrd-project/osrd/issues/6284